src/webui/clientdatastorage.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2025  Thomas Piccirello <thomas@piccirello.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "clientdatastorage.h"
+
+#include <QJsonDocument>
+#include <QJsonObject>
+
+#include "base/global.h"
+#include "base/logger.h"
+#include "base/path.h"
+#include "base/profile.h"
+#include "base/utils/io.h"
+
+const int CLIENT_DATA_FILE_MAX_SIZE = 1024 * 1024;
+const QString CLIENT_DATA_FILE_NAME = u"web_data.json"_s;
```

Comment:
Maybe be a bit more specific? For example "web_clientdata.json".

Reply:
Renamed to `web_clientdata.json`